Libraries/LibHTTP/Job.cpp

```diff
@@ -125,6 +125,17 @@ void Job::on_socket_connected()
     register_on_ready_to_read([&] {
         if (is_cancelled())
             return;
+
+        if (m_state == State::Finished) {
+            // This is probably just a EOF notification, which means we should receive nothing
+            // and then get eof() == true.
+            [[maybe_unused]] auto payload = receive(64);
+            // These assertions are only correct if "Connection: close".
+            ASSERT(payload.is_empty());
+            ASSERT(eof());
```

Comment:
We should aim towards handling this more gracefully at some point, but for now this is a progression. :)